inject-events/src/main/java/io/avaje/inject/events/spi/ObserverManagerPlugin.java

```diff
@@ -1,18 +1,19 @@
 package io.avaje.inject.events.spi;
 
 import io.avaje.inject.BeanScopeBuilder;
+import io.avaje.inject.events.ObserverManager;
 import io.avaje.inject.spi.InjectPlugin;
 
 /** Plugin for avaje inject that provides a default ObserverManager instance. */
 public final class ObserverManagerPlugin implements InjectPlugin {
 
   @Override
   public Class<?>[] provides() {
-    return new Class<?>[] {DObserverManager.class};
+    return new Class<?>[] {ObserverManager.class};
   }
 
   @Override
   public void apply(BeanScopeBuilder builder) {
-    builder.provideDefault(null, DObserverManager.class, DObserverManager::new);
+    builder.provideDefault(null, ObserverManager.class, () -> ObserverManager.builder().build());
```

Comment:
- Existing bug that was registering type DObserverManager.class ...
- Use the builder ...

To specify use of a specific Executor, wire a ObserverManager explicitly via:

```java
@Factory 
class MyFactory {
...

  @Bean
  ObserverManager observerManager(Executor executor) {
    return ObserverManager.builder()
        .asyncExecutor(exec)
        .build();
  }
}
```

... if we need to further customise the ObserverManager, we add more options to the ObserverManager.Builder in the future.